File: Constructor.py
# Python uses a class system for objects

class Car:
  __model = ''
  __year = 0
  __miles = 0

  # Constructor
  def __init__(self, model, year, miles):
    self.__model = model
    self.__year = year
    self.__miles = miles

  def toString(self):
    str = 'This {} {} has done {} miles'
    return str.format(self.__year, self.__model, self.__miles)

corolla = Car('Corolla', 2010, 80000)

print(corolla.toString())                                              

File: Constructor.js
// Javascript uses Constructor functions to create new instances of objects

function Car ( model, year, miles){
  // props and methods in the constructor are different for each instance
  this.model = model || '';
  this.year = year || 0;
  this.miles = miles || 0;
}

// Props and Methods in the prototype exist in one object and are referenced by each instance.
Car.prototype.toString = function(){
  return `This ${this.year} ${this.model} has done ${this.miles} miles`;
}

const civic = new Car ('Honda Civic', 2008, 20000);

console.log(civic.toString());